src/commcare_cloud/commands/deploy/commcare.py

```diff
@@ -1,28 +1,48 @@
+from datetime import datetime
+
+import pytz
 from clint.textui import indent
+from memoized import memoized
 
 from commcare_cloud.alias import commcare_cloud
 from commcare_cloud.cli_utils import ask
 from commcare_cloud.colors import color_summary
+from commcare_cloud.commands.deploy.utils import announce_deploy_start
+from commcare_cloud.commands.terraform.aws import get_default_username
+from commcare_cloud.commands.utils import run_fab_task
+from commcare_cloud.events import publish_deploy_event
+from commcare_cloud.fab.deploy_diff import DeployDiff
+from commcare_cloud.fab.git_repo import get_github, github_auth_provided
 
 
 def deploy_commcare(environment, args, unknown_args):
     deploy_revs, diffs = get_deploy_revs_and_diffs(environment, args)
 
-    if not confirm_deploy(deploy_revs, diffs, args):
+    if not confirm_deploy(environment, deploy_revs, diffs, args):
         return 1
 
     fab_func_args = get_deploy_commcare_fab_func_args(args)
     fab_settings = [args.fab_settings] if args.fab_settings else []
-    for name, rev in deploy_revs:
+    for name, rev in deploy_revs.items():
         var = 'code_branch' if name == 'commcare' else '{}_code_branch'.format(name)
         fab_settings.append('{}={}'.format(var, rev))
 
-    return commcare_cloud(environment.name, 'fab', 'deploy_commcare{}'.format(fab_func_args),
-                   '--set', ','.join(fab_settings),
-                   branch=args.branch, *unknown_args)
+    announce_deploy_start(environment, "CommCare HQ")
+    start = datetime.utcnow()
+    rc = commcare_cloud(
+        environment.name, 'fab', 'deploy_commcare{}'.format(fab_func_args),
+        '--set', ','.join(fab_settings), branch=args.branch, *unknown_args
+    )
+    if rc != 0:
+        return rc
```

Comment:
Is this return code used? It seems odd that the function would sometimes return a code and sometimes return `None`.
```suggestion
        return
```

Reply:
I think it should return the return code rather than None: https://github.com/dimagi/commcare-cloud/blob/6fc24a584ed2ff82bddd136be6a1526e4c8985cf/src/commcare_cloud/commcare_cloud.py#L206-L208